pygmt/tests/test_figure.py

```diff
@@ -92,6 +92,46 @@ def test_figure_savefig_exists():
         fname.unlink()
 
 
+def test_figure_savefig_geotiff():
+    """
+    Make sure .tif generates a normal TIFF file and .tiff generates a GeoTIFF
+    file.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 10, 0, 10], projection="M10c", frame=True)
+
+    # Save as GeoTIFF
+    geofname = Path("test_figure_savefig_geotiff.tiff")
+    fig.savefig(geofname)
+    assert geofname.exists()
+    # The .pgw should not exist
+    assert not geofname.with_suffix(".pgw").exists()
+
+    # Save as TIFF
+    fname = Path("test_figure_savefig_tiff.tif")
+    fig.savefig(fname)
+    assert fname.exists()
+
+    # Check if a TIFF is georeferenced or not
+    try:
+        # pylint: disable=import-outside-toplevel
+        import rioxarray
+        from rasterio.errors import NotGeoreferencedWarning
+
+        # GeoTIFF
+        with rioxarray.open_rasterio(geofname) as xds:
+            assert xds.rio.crs is not None
```

Comment:
Add some more assert statements on the lon/lat bounds and shape of the raster.

```suggestion
            assert xds.rio.crs is not None
npt.assert_allclose(
                actual=xds.rio.bounds(),
                desired=(
                    -667728.5869475466,
                    6356999.491390044,
                    594269.0244906943,
                    8381540.2498699855,
                ),
            )
            assert xds.rio.shape == (2149, 1340)
```